harper-core/src/linting/nominal_wants.rs

```diff
@@ -0,0 +1,81 @@
+use crate::{
+    Token,
+    linting::{Lint, LintKind, PatternLinter, Suggestion},
+    patterns::{Pattern, SequencePattern, WordSet},
+};
+
+pub struct NominalWants {
+    pattern: Box<dyn Pattern>,
+}
+
+impl Default for NominalWants {
+    fn default() -> Self {
+        let miss = WordSet::new(&["wont", "wonts"]);
+        let pattern = SequencePattern::default()
+            .then_nominal()
+            .then_whitespace()
+            .then(miss);
+        Self {
+            pattern: Box::new(pattern),
+        }
+    }
+}
+
+impl PatternLinter for NominalWants {
+    fn pattern(&self) -> &dyn Pattern {
+        self.pattern.as_ref()
+    }
+
+    fn match_to_lint(&self, toks: &[Token], _src: &[char]) -> Option<Lint> {
+        let offender = toks.last()?;
+        Some(Lint {
+            span: offender.span,
+            lint_kind: LintKind::Miscellaneous,
+            suggestions: vec![Suggestion::ReplaceWith("wants".chars().collect())],
+            message: "Did you mean `wants`?".to_owned(),
```

Comment:
`then_nominal` should be matching on singular and plural if plurality is propagating properly. This means either `wants` or `want` will be correct and the wrong one will introduce a different error.